src/App.tsx

```diff
@@ -1,61 +1,115 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { TodoWithoutUser, TodoWithUser, User } from './types';
+
+const preparedTodo: TodoWithUser[] = todosFromServer
+  .map((todo: TodoWithoutUser) => {
+    return {
+      ...todo,
+      user: usersFromServer
+        .find((user: User) => user.id === todo.userId) || null,
+    };
+  });
 
 export const App = () => {
+  const [visibleTodos, setVisibleTodos]
+  = useState<TodoWithUser[]>(preparedTodo);
+  const [title, setTitle] = useState('');
+  const [selecteUserId, setSelecteUserId] = useState(0);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const resetForm = () => {
+    setTitle('');
+    setSelecteUserId(0);
+    setTitleError(false);
+    setUserError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setTitleError(!title);
+    setUserError(!selecteUserId);
+
+    if (title && selecteUserId) {
+      setVisibleTodos((prevTodos) => {
+        const maxId: number = Math.max(...prevTodos.map(todo => todo.id));
+        const newTodo = {
+          id: maxId + 1,
+          title,
+          completed: false,
+          userId: selecteUserId,
+          user: usersFromServer.find(({ id }) => id === selecteUserId) || null,
+        };
+
+        return [...prevTodos, newTodo];
+      });
+      resetForm();
```

Comment:
Don't write main logic inside `if` block
Better to do
`if (!title && !selecteUserId) {
  return;
}`